authority/provisioner/dpop_options.go

```diff
@@ -0,0 +1,55 @@
+package provisioner
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"text/template"
+)
+
+type DPOPOptions struct {
+	// ValidationExecPath is the name of the executable to call for DPOP
+	// validation.
+	ValidationExecPath string `json:"validation-exec-path,omitempty"`
+	// Backend signing key for DPoP access token
+	SigningKey string `json:"key"`
+	// URI template acme client must call to fetch the DPoP challenge proof (an access token from wire-server)
+	DpopTarget string `json:"dpop-target"`
+}
+
+func (o *DPOPOptions) GetValidationExecPath() string {
+	if o == nil {
+		return "rusty-jwt-cli"
+	}
+	return o.ValidationExecPath
+}
+
+func (o *DPOPOptions) GetSigningKey() string {
+	if o == nil {
+		return ""
+	}
+	return o.SigningKey
+}
+
+func (o *DPOPOptions) GetDPOPTarget() string {
+	if o == nil {
+		return ""
+	}
+	return o.DpopTarget
+}
+
+func (o *DPOPOptions) GetTarget(deviceID string) (string, error) {
+	if o == nil {
+		return "", errors.New("misconfigured target template configuration")
+	}
+	targetTemplate := o.GetDPOPTarget()
+	tmpl, err := template.New("DeviceId").Parse(targetTemplate)
```

Comment:
I suppose sprig is not required here.

Reply:
Is this just a general remark? Doing something with the templates is one of the things that needs a bit more work, and using `sprig` could be useful to them. The goal is to dynamically get some information from the OIDC/DPoP token, and to possibly apply some transformation(s). So `sprig` support may need to be added here, I think.